private/buf/bufgen/features.go

```diff
@@ -111,9 +111,32 @@ func checkRequiredFeatures(
 				failedFeatures = append(failedFeatures, feature)
 			}
 		}
+		pluginName := configs[responseIndex].Name()
 		if supported&uint64(pluginpb.CodeGeneratorResponse_FEATURE_SUPPORTS_EDITIONS) != 0 && len(required.editionToFilenames) > 0 {
-			// Plugin supports editions, and files include editions. So make sure
-			// the plugin supports precisely the right editions.
+			// Plugin supports editions, and files include editions.
```

Comment:
What if `len(required.editionToFilenames) > 0` but the plugin response *doesn't* indicate it supports editions?

Reply:
That is already covered. This is the test case for it: https://github.com/bufbuild/buf/pull/3073/files#diff-8ab9fea8f32ab6d954c3bcfd1ca0d38d6327c57ca60e571d3d61ea914368b690R128-R134

Reply:
Missed that when looking through the code - good to know.